packages/core/src/new-api/deployer.ts

```diff
@@ -1,17 +1,69 @@
 import type { IgnitionModuleResult, ModuleParameters } from "./types/module";
 import type { IgnitionModuleDefinition } from "./types/module-builder";
 
+import { IgnitionError } from "../errors";
+
 import { Batcher } from "./internal/batcher";
-import { ExecutionEngine } from "./internal/execution-engine";
+import { ExecutionEngine } from "./internal/execution/execution-engine";
+import { BasicExecutionStrategy } from "./internal/execution/execution-strategy";
+import { executionStateReducer } from "./internal/execution/executionStateReducer";
+import { EthersChainDispatcher } from "./internal/execution/transactions/chain-dispatcher";
+import { TransactionServiceImplementation } from "./internal/execution/transactions/transaction-service";
 import { ModuleConstructor } from "./internal/module-builder";
 import { Reconcilier } from "./internal/reconciliation/reconcilier";
+import { isAdapters } from "./type-guards";
+import { Adapters } from "./types/adapters";
+import { ArtifactResolver } from "./types/artifact";
+import { DeploymentResult } from "./types/deployer";
+import { ExecutionStrategy } from "./types/execution-engine";
 import { ExecutionStateMap } from "./types/execution-state";
+import { Journal } from "./types/journal";
+import { TransactionService } from "./types/transaction-service";
 
+/**
+ * Run an Igntition deployment.
+ *
+ * @beta
+ */
 export class Deployer {
+  private _journal: Journal;
   private _moduleConstructor: ModuleConstructor;
   private _executionEngine: ExecutionEngine;
+  private _transactionService: TransactionService;
+  private _strategy: ExecutionStrategy;
+
+  constructor(
+    options:
+      | { journal: Journal; transactionService: TransactionService }
+      | {
+          journal: Journal;
+          adapters: Adapters;
+          artifactLoader: ArtifactResolver;
+        }
```

Comment:
Why the union type here? It's confusing.

Reply:
The main thing that confuses me is that they are not equivalent things.

Reply:
Have you considered creating multiple static factories and making the constructor private instead?

Reply:
Tracked by: https://github.com/NomicFoundation/ignition/issues/273